builder/docker/step_temp_dir.go

```diff
@@ -18,7 +19,23 @@ func (s *StepTempDir) Run(state multistep.StateBag) multistep.StepAction {
 	ui := state.Get("ui").(packer.Ui)
 
 	ui.Say("Creating a temporary directory for sharing data...")
-	td, err := ioutil.TempDir("", "packer-docker")
+
+	var err error
+	var td string
+	tempenv := "TMPDIR"
+	if runtime.GOOS == "windows" {
+		tempenv = "TMP"
+	}
+	if tmpdir := os.Getenv(tempenv); tmpdir != "" {
+		td, err = ioutil.TempDir(tmpdir, "packer-docker")
+	} else {
+		var configTmpDir string
+		configTmpDir, err = packer.ConfigTmpDir()
```

Comment:
Can we do the env checking in `ConfigTmpDir` so this logic doesn't need to be repeated if we want to use the tmp dir for something else?